exporters/autoexport/metrics.go

```diff
@@ -94,4 +104,62 @@ func init() {
 	RegisterMetricReader("none", func(ctx context.Context) (metric.Reader, error) {
 		return newNoopMetricReader(), nil
 	})
+	RegisterMetricReader("prometheus", func(ctx context.Context) (metric.Reader, error) {
+		// create an isolated registry instead of using the global registry --
+		// the user might not want to mix OTel with non-OTel metrics
+		reg := prometheus.NewRegistry()
+
+		mux := http.NewServeMux()
+		mux.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{Registry: reg}))
+		server := http.Server{
+			// Timeouts are necessary to make a server resilent to attacks, but ListenAndServe doesn't set any.
+			// We use values from this example: https://blog.cloudflare.com/exposing-go-on-the-internet/#:~:text=There%20are%20three%20main%20timeouts
+			// TODO: find defaults with a better justification, and/or make these configurable
+			ReadTimeout:  5 * time.Second,
+			WriteTimeout: 10 * time.Second,
+			IdleTimeout:  120 * time.Second,
+			Handler:      mux,
+		}
+
+		// environment variable names and defaults specified at https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#prometheus-exporter
+		host := getenv("OTEL_EXPORTER_PROMETHEUS_HOST", "localhost")
+		port := getenv("OTEL_EXPORTER_PROMETHEUS_PORT", "9464")
+		addr := host + ":" + port
+		lis, err := net.Listen("tcp", addr)
+		if err != nil {
+			// TODO: retry binding, to handle process restarts
+			otel.Handle(fmt.Errorf("unable to bind address %s for Prometheus exporter: %w", addr, err))
+		}
+
+		go func() {
+			otel.Handle(server.Serve(lis))
```

Comment:
From: https://pkg.go.dev/net/http#Server.Serve

> Serve always returns a non-nil error and closes l. After Shutdown or Close, the returned error is ErrServerClosed.

I do not think we want to report `ErrServerClosed`.

I think may be good to verify in the tests that no errors was reported via `otel.Handle`. You can do it by using `otel.SetErrorHandler`.

Reply:
Done.